app/src/androidTest/java/com/github/steroidteam/todolist/CustomMatchers.java

```diff
@@ -0,0 +1,90 @@
+import java.util.Objects;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.core.IsAnything;
+
+public class CustomMatchers {
+    /**
+     * Helper to check if the given text is the same as the to-do list title at the given position
+     *
+     * @param position the index of the to-do in the recycler view
+     * @param expectedText the text that should be the title of the to-do
+     * @return Mactcher to test the title
+     */
+    public static Matcher<View> atPositionCheckText(
+            final int position, @NonNull final String expectedText, @NonNull final int layout_id) {
+        return new BoundedMatcher<View, RecyclerView>(RecyclerView.class) {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText(
+                        "View holder at position "
+                                + position
+                                + ", expected: "
+                                + expectedText
+                                + " ");
+            }
+
+            @Override
+            protected boolean matchesSafely(final RecyclerView view) {
+                View holder = view.getChildAt(position);
+                TextView bodyView = holder.findViewById(layout_id);
+                return bodyView.getText().toString().equals(expectedText);
+            }
+        };
+    }
+
+    /** Helper to check the size of a recyclerView */
+    public static Matcher<View> ItemCountIs(@NonNull final int expectedCount) {
+        return new BoundedMatcher<View, RecyclerView>(RecyclerView.class) {
+            @Override
+            public void describeTo(Description description) {
+                description.appendText("Recycler View should have " + expectedCount + " items");
+            }
+
+            @Override
+            protected boolean matchesSafely(final RecyclerView view) {
+                return Objects.requireNonNull(view.getAdapter()).getItemCount() == expectedCount;
+            }
+        };
+    }
+
+    /**
+     * Helper to test dialog
+     *
+     * @param millis time to wait
+     * @return viewAction that wait the given time
+     */
+    public static ViewAction waitAtLeastHelper(final long millis) {
+        return new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return anyView();
+            }
+
+            @NonNull
+            private Matcher<View> anyView() {
+                return new IsAnything<>();
+            }
+
+            @Override
+            public String getDescription() {
+                return "wait for at least " + millis + " millis.";
+            }
+
+            @Override
+            public void perform(final UiController uiController, final View view) {
+                uiController.loopMainThreadUntilIdle();
+                uiController.loopMainThreadForAtLeast(millis);
+            }
+        };
+    }
```

Comment:
We have a separate directory for custom `ViewAction`s, see `com.github.steroidteam.todolist.customviewactions`.

Reply:
Maybe you can add the custom `ViewAction` as a class there, and then create `waitAtLeastHelper` as a static method.

Reply:
In fact, I just realized that this helper is not used at all, so I am going to remove it in an additional commit.